plugins/mimecast/komand_mimecast/tasks/monitor_siem_logs/task.py

```diff
@@ -76,6 +112,16 @@ def run(  # pylint: disable=unused-argument # noqa: MC0001
                     self.logger.info("Caught up, saving 'normal_running_cutoff' as 'True' to the state")
                     state["normal_running_cutoff"] = True
 
+            state[self.LAST_LOG_LINE] = last_log_line
+
+            if last_log_line > 0:
+                state[self.LAST_RUNS_FILTER_TIME] = filter_time.isoformat()
+                state[self.PREVIOUS_FILE_HASH] = current_file_hash
```

Comment:
hopefully unlikely to happen, but if we run the same header token and get different files which could be less than `max_events_per_run`, current_file_hash will throw variable undefined. could just set as None at the start of the task

Reply:
this has been covered by changing the above if statment to check for a last file hash in the state